### PR TITLE
improve test case coverage; fix Utility.encode bug

### PR DESCRIPTION
**Pull request by markro49**
1) add test code for BCEL-340.
2) add new test to improve code coverage for classfile/Utility.java.
coverage increased from 40% to 72%.
3) fix bug in encode that (2) exposed.

### CODE REVIEW

File: src/test/java/org/apache/bcel/data/DummySignatureParser.java [hunk 1]
```diff
@@ -0,0 +1,1049 @@
...
+
+package org.apache.bcel.data;
+
+/*
+ * The following code is taken from the JDK 8 sources for sun.reflect.generics.
+ * It has been modified to compile stand alone within the BCEL test environment.
+ * The code is not executed but the classfile is used as input to a BCEL test case.
+ */
+
+/*
```
Comment (garydgregory) [inline src/test/java/org/apache/bcel/data/DummySignatureParser.java:27]: -1: We cannot include GPL 2 or GPL 3 code. Please see:
- https://www.apache.org/licenses/GPL-compatibility.html
- https://www.apache.org/foundation/license-faq.html#GPL

**Comment by markro49** [comment]
Oh rats - I will have to spend time writing my own comprehensive test case.

**Comment by garydgregory** [comment]
> Oh rats - I will have to spend time writing my own comprehensive test case.

Yeah, it's pain, sorry about that, but there's not much we can do about GPL.

**Comment by garydgregory** [comment]
@markro49 ping?

**Pull request merged**